File: Loesungen/loesung_10.py
from library import *


if __name__ == "__main__":
    print("n  bitpa encbp eVal dVal")
    for n in range(0,32):
        encg = gray_encode(n)
        decg = gray_decode(encg)
        nStr = to_binary(n)
        nStrToAsci = "%02d" % n
        encgToAsci = "%04d" % encg
        decgStrToAsci = ("%04d" % decg)
        res=("%02d "%n + nStr+" "+to_binary(n)+" "+to_binary(encg)+" "+encgToAsci+ " "+ decgStrToAsci)
        print(res)        

File: Loesungen/loesung_10.cpp
// Bestimmt das kleinste gemeinsame Vielfachte (kgv)
// basic file operations
#include <iostream>
#include  <vector>
#include "library.h"

//#include "version.h"
using namespace std;


int main (int argc, char *argv[]) {
	std::cout << "Number\tBinary\tGray\tDecoded" << std::endl;
	std::cout << "------\t------\t----\t-------" << std::endl;
	for (unsigned int n=0;n<32;++n)
	{
		auto encg = gray_encode(n);
		auto decg = gray_decode(encg);
		std::cout << n  << "\t" << to_binary(n) << "\t" << to_binary(encg) << "\t" << decg << std::endl;
	}
		
    return 0;
}